userspace/otpilot/src/main.rs

```diff
@@ -85,7 +85,7 @@ fn run() -> TockResult<()> {
     {
         let mut idx : usize = 0;
         for val in "v1.00".as_bytes() {
-            if idx > identity.version.len() { break; }
+            if idx >= identity.version.len() { break; }
             identity.version[idx] = *val;
             idx = idx + 1;
         }
```

Comment:
I know I don't know rust, but this seems like a really weird way to write this loop, and I'm not all surprised that this bug was written as a result.
Is this really the right way to iterate through the bytes that make up a string?

Reply:
I only found this bug in a later version of this code where the "v1.00" is NOT hardcoded but derived from the GIT commit ID at build time. 

But yeah, that code isn't good. Please have a look at the rewrite.